Source/Diagnostics/BoostedFrameDiagnostic.H

```diff
@@ -20,55 +20,123 @@
+    std::unique_ptr<amrex::MultiFab> data_buffer_;
+    // particles_buffer_ is currently blind to refinement level.
+    // particles_buffer_[j] is a WarpXParticleContainer::DiagnosticParticleData where
+    // - j is the species number for the current diag
+    amrex::Vector<WarpXParticleContainer::DiagnosticParticleData> particles_buffer_;
+    // buff_counter_ is the number of z slices in data_buffer_
+    int buff_counter_;
+    int num_buffer_ = 256;
+    int max_box_size = 256;
     ///
-    /// LabSnapShot stores metadata corresponding to a single time
-    /// snapshot in the lab frame. The snapshot is written to disk
-    /// in the directory "file_name". zmin_lab, zmax_lab, and t_lab
-    /// are all constant for a given snapshot. current_z_lab and
-    /// current_z_boost for each snapshot are updated as the
-    /// simulation time in the boosted frame advances.
+    /// This snapshot is at time t_lab, and the simulation is at time t_boost.
+    /// The Lorentz transformation picks out one slice corresponding to both
+    /// of those times, at position current_z_boost and current_z_lab in the
+    /// boosted and lab frames, respectively.
     ///
-    struct LabSnapShot {
-
-        std::string file_name;
-        amrex::Real t_lab;
-        amrex::RealBox prob_domain_lab_;
-        amrex::IntVect prob_ncells_lab_;
-
-        amrex::Real current_z_lab;
-        amrex::Real current_z_boost;
-
-        int ncomp_to_dump_;
-        std::vector<std::string> mesh_field_names_;
-
-        int file_num;
-        int initial_i;
-        const BoostedFrameDiagnostic& my_bfd;
-
-        LabSnapShot(amrex::Real t_lab_in, amrex::Real t_boost,
-                    const amrex::RealBox prob_domain_lab,
-                    const amrex::IntVect prob_ncells_lab,
-                    const int ncomp_to_dump,
-                    const std::vector<std::string> mesh_field_names,
-                    int file_num_in,
-                    const BoostedFrameDiagnostic& bfd);
-
-        ///
-        /// This snapshot is at time t_lab, and the simulation is at time t_boost.
-        /// The Lorentz transformation picks out one slice corresponding to both
-        /// of those times, at position current_z_boost and current_z_lab in the
-        /// boosted and lab frames, respectively.
-        ///
-        void updateCurrentZPositions(amrex::Real t_boost, amrex::Real inv_gamma,
-                                     amrex::Real inv_beta);
-
-        ///
-        /// Write some useful metadata about this snapshot.
-        ///
-        void writeSnapShotHeader();
-    };
+    void updateCurrentZPositions(amrex::Real t_boost, amrex::Real inv_gamma,
+                                 amrex::Real inv_beta);
+
+    void createLabFrameDirectories();
+    ///
+    /// Write some useful metadata about this snapshot.
+    ///
+    void writeLabFrameHeader();
+
+    virtual int getId() {return 0;}
```

Comment:
There are a number of member functions and variables that are not used. For instance, I could not find any call to `getId`, and any place where `snapshot_id` and `slice_id` are used. Did I miss something? If not, could you do some cleaning and remove *all* the unused members? Compiling with `DEBUG=TRUE` might help, but I am not sure it catches everything.

Reply:
Thank you for this suggestion. I have now removed unused variables and functions detected using the DEBUG flag.